map/src/menu/tracks/CloudTrackItem.jsx

```diff
@@ -1,24 +1,64 @@
 import AppContext, { OBJECT_TYPE_CLOUD_TRACK } from '../../context/AppContext';
-import { Alert, LinearProgress, ListItemText, MenuItem, Switch, Tooltip, Typography } from '@mui/material';
-import { useContext, useState, useMemo, useEffect } from 'react';
-import Utils from '../../util/Utils';
+import {
+    Alert,
+    ClickAwayListener,
+    IconButton,
+    LinearProgress,
+    ListItemIcon,
+    ListItemText,
+    MenuItem,
+    Popper,
+    Switch,
+    Tooltip,
+    Typography,
+} from '@mui/material';
+import React, { useContext, useEffect, useMemo, useRef, useState } from 'react';
+import Utils, { toHHMMSS } from '../../util/Utils';
 import TrackInfo from './TrackInfo';
 import TracksManager, { isEmptyTrack } from '../../manager/TracksManager';
 import _ from 'lodash';
 import { useWindowSize } from '../../util/hooks/useWindowSize';
+import { ReactComponent as TrackIcon } from '../../assets/icons/ic_action_polygom_dark.svg';
+import { ReactComponent as MenuIcon } from '../../assets/icons/ic_overflow_menu_white.svg';
+import { ReactComponent as MenuIconHover } from '../../assets/icons/ic_overflow_menu_with_background.svg';
+import styles from './trackmenu.module.css';
+import TrackActions from './actions/TrackActions';
 
-export default function CloudTrackItem({ file, customIcon = null }) {
+export default function CloudTrackItem({ file, customIcon = null, visible = null }) {
     const ctx = useContext(AppContext);
 
     const [, , mobile] = useWindowSize();
 
     const [loadingTrack, setLoadingTrack] = useState(false);
     const [error, setError] = useState('');
+    const [hoverIconInfo, setHoverIconInfo] = useState(false);
+    const [showTrack, setShowTrack] = useState(false);
+    const [openActions, setOpenActions] = useState(false);
+    const anchorEl = useRef(null);
 
     const info = useMemo(() => <TrackInfo file={file} />, [file]);
 
+    const dist = getDist(file);
+    const time = getTime(file);
+    const points = getPoints(file);
+
     const [displayTrack, setDisplayTrack] = useState(null); // null -> true/false -> null
 
+    function getDist(file) {
+        let f = file.details ? file.details : file;
+        return f?.analysis?.totalDistance ? (f?.analysis?.totalDistance / 1000).toFixed(2) : 0;
+    }
+
+    function getTime(file) {
+        let f = file.details ? file.details : file;
+        return f?.analysis?.timeMoving ? toHHMMSS(f?.analysis?.timeMoving) : '0:00';
+    }
+
+    function getPoints(file) {
+        let f = file.details ? file.details : file;
+        return f?.analysis?.points ? f?.analysis?.points : 0;
+    }
+
```

Comment:
Better to use empty default values instead of "0" / "0:00" to avoid ugly "00:000" text on tracks w/o any details